bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WindowsDefenderConfigurator.java

```diff
@@ -173,7 +175,7 @@ private static Boolean runExclusionCheck(IProgressMonitor m, Optional<Path> inst
 	private static HandlingOption askForDefenderHandlingDecision() {
 		String message = bindProductName(WorkbenchMessages.WindowsDefenderConfigurator_exclusionCheckMessage);
 
-		return PlatformUI.getWorkbench().getDisplay().syncCall(() -> {
+		return getDisplay().syncCall(() -> {
```

Comment:
here you fetch the display, why not use it below for the dialog as well and instead fetch it again.
Beside that if you want the Shell, your probably should inject that as well...

Reply:
Good point about the double fetching, reused that.